ibm_cloud_sdk_core/iam_token_manager.py

```diff
@@ -39,40 +58,32 @@ def request_token(self):
             'Content-type': self.CONTENT_TYPE,
             'Accept': 'application/json'
         }
+        if self.headers is not None and isinstance(self.headers, dict):
+            headers.update(self.headers)
+
         data = {
             'grant_type': self.REQUEST_TOKEN_GRANT_TYPE,
-            'apikey': self.iam_apikey,
+            'apikey': self.apikey,
             'response_type': self.REQUEST_TOKEN_RESPONSE_TYPE
         }
 
         # Use bx:bx as default auth header creds
         auth_tuple = ('bx', 'bx')
 
         # If both the clientId and secret were specified by the user, then use them
-        if self.iam_client_id and self.iam_client_secret:
-            auth_tuple = (self.iam_client_id, self.iam_client_secret)
+        if self.client_id and self.client_secret:
+            auth_tuple = (self.client_id, self.client_secret)
```

Comment:
The new behavior is to just not send an auth header if the client id and secret are not set, rather than defaulting to `bx:bx`.